### DISCUSSION TITLE
What's the propagation length of the simulation of bending fiber in your paper"BPM-Matlab: an open-source optical propagation simulation tool in MATLAB"

### DISCUSSION THREAD

**Discussion by LuoXiaobupt** [ankrh/BPM-Matlab#1] [General]
Dear ankrh:
BPM-matlab is a great simulation tool for fibre modeing! 
Could you please tell me the propagation length you set in the simulation of bending fibre in your paper "BPM-Matlab: an open-source optical propagation simulation tool in MATLAB"？
I would appreciate it if you could reply to me！

**Comment by ankrh** [comment]
Dear Luo Xiao

I’m afraid I don’t have the original simulation model files anymore and I don’t remember exactly which propagation length I used. But I remember that I plotted the loss per unit length as function of propagation distance.

When injecting the mode I wanted to study, there were some transient fluctuations where the loss per unit distance was not yet stable, but after a certain distance it approached a certain limit value, which is what I put into the article.

The required distances were not very large since the limit value was reached rather quickly and the simulations just took a few seconds.

BR Anders


From: LuoXiaobupt ***@***.***>
Sent: 23. maj 2022 15:46
To: ankrh/BPM-Matlab ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ankrh/BPM-Matlab] What's the propagation length of the simulation of bending fiber in your paper"BPM-Matlab: an open-source optical propagation simulation tool in MATLAB" (Discussion #1)


Dear ankrh:
BPM-matlab is a great simulation tool for fibre modeing!
Could you please tell me the propagation length you set in the simulation of bending fibre in your paper "BPM-Matlab: an open-source optical propagation simulation tool in MATLAB"？
I would appreciate it if you could reply to me！

**Reply by LuoXiaobupt** [reply]
Dear BR Anders

I'm very glad you can reply to me. And I want to ask you for more details

The BPM-Matlab results for bent are agree with Ref.41, where the propagation distance is also not described in the simulation of bending optical fiber. What is used to measure consistency?

Additionally, In Ref.41, it is mentioned that the refractive index was truncated for larger values of x in order to prevent loss to the cladding during the mode calculations of a bending fibre. Did BPM-Matlab do the same as Ref.41? If so, how is the value of x_trans set?

Sincerely
Luo Xiao

**Comment by ankrh** [comment]
Dear Luo Xiao

The propagation length should just be long enough for any transients to smooth out and you get a constant loss per unit length of propagation. As long as you propagate that far, it doesn’t matter how long the propagation distance is. If you plot relative power loss per unit length propagation as function of the distance of propagation, you should easily be able to see the transients die out and the loss stabilize. I’m sorry that I don’t have the scripts anymore to know which lengths were relevant for the data in the article.

The way I did it was simply to make the window size only just as large as necessary to contain the light, and no larger. This functions equivalently to truncating the refractive index as they did in the reference. The window size was set via trial and error. If the window is too large, one can clearly see the modes found with findModes become unphysical and get an exponential growth in the direction away from the bend.

Best regards
Anders Kragh Hansen

**Comment by LuoXiaobupt** [comment]
Dear Anders Kragh Hansen 

I'm so sorry I made a mistake in addressing you.

I got constructive inspiration from your reply and the problem that bothered me for a long time was finally solved.

At present, I have only one doubt left. Mathematical description of bending involves with parameters including Poisson ratio and photo-elastic components (P11 and P12), but what's troubling is that the value is not specified in ref.41 and its references, could you please tell me what you set the values of these parameters?

Sincerely
Luo Xiao

**Comment by ankrh** [comment]
Hi Madhu
(Luo Xiao cc)

Do you remember where we got the rho_e = 0.22 value that we put into BPM-Matlab? I think it was from a reference you found originally, but apparently not the one we referenced in our article.

BR Anders

**Comment by ankrh** [comment]
Hi Anders,

The value of rho_e = 0.22 was adapted from Esben’s thesis, which is available here: https://dtudk.sharepoint.com/:b:/r/sites/BiomedicalOptics2/Delte%20dokumenter/proj%20T-SPIF/Esben_Andresen_Lille_Accreditation_v5.pdf?csf=1&web=1&e=Ifmj1v, Eqn. (5.8).

They have also discussed it in their publications:
1. https://doi.org/10.1364/OE.25.031863, Eqn. (1)
2. https://doi.org/10.1364/OPTICA.6.001185, supplementary material, Eqn. (S12).

Best regards,
Madhu Veettikazhy
Postdoc


DTU Health Tech
https://www.healthtech.dtu.dk/biophotonic-imaging



From: Anders Kragh Hansen ***@***.***>
Sent: 30. maj 2022 13:23
To: Madhu Veettikazhy ***@***.***>
Cc: ankrh/BPM-Matlab ***@***.***>
Subject: RE: [ankrh/BPM-Matlab] What's the propagation length of the simulation of bending fiber in your paper"BPM-Matlab: an open-source optical propagation simulation tool in MATLAB" (Discussion #1)


Hi Madhu
(Luo Xiao cc)

Do you remember where we got the rho_e = 0.22 value that we put into BPM-Matlab? I think it was from a reference you found originally, but apparently not the one we referenced in our article.

BR Anders

**Comment by LuoXiaobupt** [comment]
Dear Anders Kragh Hansen

Do you mean that you write Eq.8 in your paper"BPM-Matlab......" as the follow formula  by substituting the value of the parameter?

n'(x, y) = n(x, y)[1-0.22*x/R]exp(x/R)

Sincerely 
Luo Xiao

**Comment by ankrh** [comment]
There is an n^2 missing from your expression but otherwise yes, that’s how the 0.22 is used.
***@***.***

See also line 473 in FDBPMpropagator.c, which is where the calculation actually happens:
    float n_bend = CREALF(n)*(1-(sqrf(CREALF(n))*(x*P->cosBendDirection+y*P->sinBendDirection)/2/P->RoC*P->rho_e))*exp((x*P->cosBendDirection+y*P->sinBendDirection)/P->RoC);

Where P->rho_e is the photoelastic coefficient, P->RoC is the radius of curvature CREALF is a function that just returns the real part of the complex refractive index, sqrf() is a function that squares the input, and P->cosBendDirection is 1 for bending in the x direction, while P->sinBendDirection is 0.

BR Anders

**Comment by LuoXiaobupt** [comment]
Dear Anders Kragh Hansen

In the scalar paraxial Helmholtz equation, n0 represents the reference refractive index, and the product of k0 and n0 represents reference propagation constant. In fact, different modes have different propagation constants. In my simulation, using other method, the input is a guided mode, and once the reference propagation constant deviates from the exact value, the amplitude of the mode field will gradually decay with the increase of the propagation distance.

In BPM-matlab, it seems that the value of n0 only changes the speed of phase change. And in function Findmodes, only the effective refractive index that less than n0 can be found. Could please tell me what other role does a play in your design? 

For the same wavelength, if the input is not a single guided mode, but the sum of multiple guided modes, how should the propagation constant be set? If we choose a reference propagation constant that is chosen to represent the average phase variation of the field, will it cause amplitude attenuation of different guided mode components?

Sincerely
Luo Xiao

**Comment by ankrh** [comment]
Dear Luo Xiao

We haven’t performed a study to test the consequences of choosing n0 very different to the effective refracting indices of the propagating modes, so I can’t say for sure what the impact will be. But n0 appear as a term in the equation to minimize the possibility of numerical errors resulting from very fast phase changes.

I can’t say why the other method you use has a gradually decaying amplitude if you choose a wrong n0 –that sounds like a bug. I don’t think BPM-Matlab will have this behavior, but the results may nevertheless still become a bit less accurate.

The best thing for you to do is to test various values of n0 and see how much the end result changes.

When propagating multiple modes, I would personally just set n0 to be somewhere in between the effective refractive indices of the modes.

BR Anders

**Reply by LuoXiaobupt** [reply]
Dear Anders Kragh Hansen

As is known, the propagation constant of the mode is formed by the product of the effective refractive index of the mode and the vacuum wave number k0. Light of one frequency are incident into the fiber at different angles to form different modes, and each mode has a different axial velocity, i.e., the propagation constant, thus resulting in the dispersion of modes. Hyperbolic secant graded index fiber is designed to completely eliminate modal dispersion, and this is to say all of the guided-modes should have the same propagation constant, but when I call the method "findmode" in BPM-matlab for a hyperbolic secant graded index fiber, the calculated guided modes include LP01, LP11, LP21 still have different effective refractive index. 

Could you please tell me why this happen? Is there a wrong with my understanding of the effective refractive index?

The following is the corresponding code，and in the part of refractive index setting, the refractive index profiles corresponding to step fiber and hyperbolic secant graded index fiber are given respectively.
%%%%%%%%%%%%%%%%%%%%%%%%%
clear P % Parameters struct
% This example shows a basic case of launching an off-center Gaussian beam
% into a multimode step index fiber core and watching as the modes beat
% with each other.

%% General and solver-related settings
P.name = mfilename;
P.useAllCPUs = false; % If false, BPM-Matlab will leave one processor unused. Useful for doing other work on the PC while simulations are running.
P.useGPU = false; % (Default: false) Use CUDA acceleration for NVIDIA GPUs

%% Visualization parameters
P.updates = 10;            % Number of times to update plot. Must be at least 1, showing the final state.
P.plotEmax = 1; % Max of color scale in the intensity plot, relative to the peak of initial intensity
P.storeE3D = true;
% P.calcModeOverlaps = true;  

%% Resolution-related parameters (check for convergence)
P.Lx_main = 50e-6;        % [m] x side length of main area
P.Ly_main = 50e-6;        % [m] y side length of main area
P.Nx_main = 101;          % x resolution of main area
P.Ny_main = 101;          % y resolution of main area
P.padfactor = 1.0;  % How much absorbing padding to add on the sides of the main area (1 means no padding, 2 means the absorbing padding on both sides is of thickness Lx_main/2)
P.dz_target = 0.5e-6; % [m] z step size to aim for
P.alpha = 3e14;             % [1/m^3] "Absorption coefficient" per squared unit length distance out from edge of main area

%% Problem definition
P.lambda = 1064e-9; % [m] Wavelength
P.n_background = 1.52; % [] (may be complex) Background refractive index (in this case, the cladding)
P.n_0 = sqrt(1.52^2+0.01);

P.n.func = @calcRI;

P = findModes(P,10);
%% USER DEFINED RI FUNCTIONS
function n = calcRI(X,Y,n_background,nParameters)

% % step index fiber
% n = n_background*ones(size(X)); % Start by setting all pixels to n_background  [101, 101]
% n(X.^2 + Y.^2 < 12.5e-6^2) = sqrt(1.52^2+0.01); 

% Hyperbolic secant graded index fiber
n1 = sqrt(1.52^2+0.01);
n2 = 1.52;
a = 12.5e-6;
dete = (n1^2-n2^2)/(2*n1^2);
% n may be complex
n = n_background*ones(size(X)); % Start by setting all pixels to n_background  [101, 101]
R = sqrt(X.^2+Y.^2);
n(R<a) = n1*sech(sqrt(2*dete)*R(R<a)/a);
 
end
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Sincerely look forward to your reply！
Luo Xiao
